packages/e2e-tests/helpers/screenshot.ts

```diff
@@ -66,9 +70,54 @@ export async function hoverScreenshot(page: Page, xPercentage: number, yPercenta
   await screenshot.hover({ position, force: true });
 }
 
-export async function getGraphicsDataUrl(page: Page): Promise<string> {
+export async function getGraphicsDataUrl(page: Page): Promise<string | null> {
   return await page.evaluate(() => {
     const element = document.querySelector("#graphics") as HTMLImageElement;
-    return element.src;
+    return element?.src ?? null;
   });
 }
+
+export async function getGraphicsExecutionPoint(page: Page): Promise<string | null> {
+  const element = getVideoElement(page);
+  return await element.getAttribute("data-execution-point");
+}
+
+export async function getGraphicsStatus(page: Page): Promise<string | null> {
+  const element = getVideoElement(page);
+  return await element.getAttribute("data-status");
+}
+
+export async function getGraphicsTime(page: Page): Promise<number | null> {
+  const element = getVideoElement(page);
+  const value = await element.getAttribute("data-time");
+  return value != null ? parseInt(value) : null;
+}
+
+export async function getGraphicsPixelColor(page: Page, x: number, y: number) {
```

Comment:
This helper function isn't _necessary_ but it makes `expect` failures small and easier to read than if we were comparing the full data url.